Add API Environment management documentation (#69)
ing-bank/ingenious-doc: docs/api/apiworkbench.md

Reviewer: Update the screenshot. Note that the API Workbench button now looks like this 

<img width="222" height="78" alt="Image" src="https://github.com/user-attachments/assets/8ce62d3c-f0b5-4873-b650-8215dad125d9" />

Reviewer: same as previous comment

Reviewer: Add a line before this that cues the reader that the numbers in the previous image correspond to the pages below. Something in the lines of "Usage of each of the numbered sections above are described further in the following tabbed pane:" Improve the wording as you see fit.

=== ":three: Response Body Pane"

>       This is section displays the API response content in a formatted (Pretty) or raw view, allowing you to easily inspect, copy, and analyze returned data for validation and troubleshooting.

Reviewer 2: ```suggestion
      This section displays the API response content in a formatted (Pretty) or raw view, allowing you to easily inspect, copy, and analyze returned data for validation and troubleshooting.
```

=== ":four: API Environments"

      API Environments allow you to store reusable variables for different testing contexts such as Development, QA, Staging, or Production.

      Variables can be referenced using the syntax:

      ```text
      {{variableName}}
      ```

      This allows the same request to be reused across multiple environments without manually changing values such as URLs, tokens, usernames, passwords, proxy settings, or certificate paths.

      **Environment List**

      ![APIEnvironmentList](../img/api/APIEnvironmentList.png "APIEnvironmentList")

      **Environment Variable Configuration**

      ![APIEnvironmentVariables](../img/api/APIEnvironmentVariables.png "APIEnvironmentVariables")

      **Request Using Variables and Response**

      ![APIEnvironmentRequestExample](../img/api/APIEnvironmentRequestExample.png "APIEnvironmentRequestExample")

      When a request is executed, all referenced variables are automatically resolved using the currently selected environment.

> === ":five: API Request Conversion to Test Case"

Reviewer 3: ```suggestion
=== ":five: Convert API Request to Test Case"
```

=== ":four: API Environments"

      API Environments allow you to store reusable variables for different testing contexts such as Development, QA, Staging, or Production. This enables the same API request to be reused across multiple environments without manually changing values such as URLs, tokens, usernames, passwords, proxy settings, or certificate paths.

      The **Environment Dropdown** allows you to switch between environments. To open the **Environment Configuration Window** click **Configure**.

      ![APIEnvironmentDropdown](../img/api/APIEnvironmentDropdown.png "APIEnvironmentDropdown")

      The Environment Configuration Window displays all available environments and provides several actions:

      * **+ New** – Create a new environment.
      * **Rename** – Change the name of the selected environment.
      * **Duplicate** – Create a copy of the selected environment, including all configured variables.
      * **Delete** – Remove the selected environment.

      ![APIEnvironmentActions](../img/api/APIEnvironmentActions.png "APIEnvironmentActions")

      **Environment Variable Configuration**

>       Variables are configured by entering a **Name** and **Value**. The **Secret** option can be enabled for sensitive information such as passwords, tokens, and API keys. After adding or updating variables, click **Save** to persist the environment configuration.

Reviewer 3: ```suggestion
      Variables are configured by entering a **Name** and **Value**. The **Secret** option can be enabled for sensitive information such as passwords, tokens, and API keys. Secret values are masked in the UI and are not displayed in plain text. After adding or updating variables, click **Save** to persist the environment configuration.
```

      ![APIEnvironmentVariables](../img/api/APIEnvironmentVariables.png "APIEnvironmentVariables")

      Variables can be referenced throughout API requests using double curly braces:

      ```text
      {{variableName}}
      ```

      For example:

      ```text
      {{baseUrl}}/users
      ```

      ![APIEnvironmentRequestExample](../img/api/APIEnvironmentRequestExample.png "APIEnvironmentRequestExample")

>       When a request is executed, all referenced variables are automatically resolved using the currently selected environment. Switching to a different environment allows the same request to be executed against a different set of values without modifying the request itself.

Reviewer 3: ```suggestion
      When a request is executed, all referenced variables are automatically resolved using the currently selected environment. Switching to a different environment allows the same request to be executed against a different set of values without modifying the request itself. If a referenced variable does not exist in the active environment, the request execution may fail or use the unresolved placeholder value
```

> 

Reviewer 3: Sugggestion: Add tip box or note box

**Best Practice**
    Store environment-specific values such as URLs, credentials, API keys, and tokens as environment variables instead of hardcoding them in requests. This improves maintainability and portability.

Author: Tip and note boxes don't work since it's already under a tab, so I just added it as normal text.

=== ":five: Convert API Request to Test Case"

      API requests can be converted directly into INGenious test cases, allowing API validations to be incorporated into automated test suites with minimal effort.

      During conversion, all configured request details are preserved, including:

      * Endpoint URL and query parameters
      * Request headers
      * Authentication settings (Basic Auth, Bearer Token, and API Keys)
      * Request body
      * Assertions

>       If the request contains environment variables (for example `{{baseUrl}}` or `{{token}}`), the values are resolved using the **currently active environment** during conversion. If regular text values are used instead of variables, those values are converted as-is.

Reviewer 3: ```suggestion
      If the request contains environment variables (for example `{{baseUrl}}` or `{{token}}`), the values are resolved using the **currently active environment** during conversion. Literal values that do not reference environment variables are copied directly into the generated test case.
```
